exchanges/coinbasepro/coinbasepro_types.go

```diff
@@ -1,518 +1,1525 @@
+	CreatedAt        time.Time `json:"created_at"`
+	UpdatedAt        time.Time `json:"updated_at"`
+	DeletedAt        time.Time `json:"deleted_at"`
+	Type             string    `json:"type"`
+	Ready            bool      `json:"ready"`
+	Hold             ValCur    `json:"hold"`
+}
+
+// AllAccountsResponse holds many Account structs, as well as pagination information,
+// returned by GetAllAccounts
+type AllAccountsResponse struct {
+	Accounts []Account `json:"accounts"`
+	HasNext  bool      `json:"has_next"`
+	Cursor   string    `json:"cursor"`
+	Size     uint8     `json:"size"`
+}
+
+// Params is used within functions to make the setting of parameters easier
+type Params struct {
+	url.Values
+}
+
+// PriceSize is a sub-struct used in the type ProductBook
+type PriceSize struct {
+	Price float64 `json:"price,string"`
+	Size  float64 `json:"size,string"`
+}
+
+// ProductBook holds bid and ask prices for a particular product, returned by GetProductBookV3
+type ProductBook struct {
+	ProductID currency.Pair `json:"product_id"`
+	Bids      []PriceSize   `json:"bids"`
+	Asks      []PriceSize   `json:"asks"`
+	Time      time.Time     `json:"time"`
+}
+
+// FCMTradingSessionDetails is a sub-struct used in the type Product
+type FCMTradingSessionDetails struct {
+	IsSessionOpen                bool      `json:"is_session_open"`
+	OpenTime                     time.Time `json:"open_time"`
+	CloseTime                    time.Time `json:"close_time"`
+	SessionState                 string    `json:"session_state"`
+	AfterHoursOrderEntryDisabled bool      `json:"after_hours_order_entry_disabled"`
+}
+
+// PerpetualDetails is a sub-struct used in the type FutureProductDetails
+type PerpetualDetails struct {
+	OpenInterest  types.Number `json:"open_interest"`
+	FundingRate   types.Number `json:"funding_rate"`
+	FundingTime   time.Time    `json:"funding_time"`
+	MaxLeverage   types.Number `json:"max_leverage"`
+	BaseAssetUUID string       `json:"base_asset_uuid"`
+}
+
+// FutureProductDetails is a sub-struct used in the type Product
+type FutureProductDetails struct {
+	Venue                  string           `json:"venue"`
+	ContractCode           string           `json:"contract_code"`
+	ContractExpiry         time.Time        `json:"contract_expiry"`
+	ContractSize           types.Number     `json:"contract_size"`
+	ContractRootUnit       string           `json:"contract_root_unit"`
+	GroupDescription       string           `json:"group_description"`
+	ContractExpiryTimezone string           `json:"contract_expiry_timezone"`
+	GroupShortDescription  string           `json:"group_short_description"`
+	RiskManagedBy          string           `json:"risk_managed_by"`
+	ContractExpiryType     string           `json:"contract_expiry_type"`
+	PerpetualDetails       PerpetualDetails `json:"perpetual_details"`
+	ContractDisplayName    string           `json:"contract_display_name"`
+	TimeToExpiryMS         uint64           `json:"time_to_expiry_ms,string"`
```

Comment:
time.Duration?

Reply:
I can't actually test whether that's valid, but I can plug it in anyway.